wandb: WARNING Using legacy-service, which is deprecated. If this is unintentional, you can fix it by ensuring you do not call `wandb.require('legacy-service')` and do not set the WANDB_X_REQUIRE_LEGACY_SERVICE environment variable.
wandb: Starting wandb agent üïµÔ∏è
2025-04-23 16:27:40,716 - wandb.wandb_agent - INFO - Running runs: []
2025-04-23 16:27:41,459 - wandb.wandb_agent - INFO - Agent received command: run
2025-04-23 16:27:41,459 - wandb.wandb_agent - INFO - Agent starting run with config:
	batch_size: 128
	learning_rate: 0.00016624389245449225
	num_residual_blocks: 16
	residual_scaling: 0.3
2025-04-23 16:27:41,466 - wandb.wandb_agent - INFO - About to run command: /usr/bin/env python src/optimize_wandb.py --batch_size=128 --learning_rate=0.00016624389245449225 --num_residual_blocks=16 --residual_scaling=0.3
2025-04-23 16:27:46,483 - wandb.wandb_agent - INFO - Running runs: ['a79wjaye']
wandb: WARNING Using legacy-service, which is deprecated. If this is unintentional, you can fix it by ensuring you do not call `wandb.require('legacy-service')` and do not set the WANDB_X_REQUIRE_LEGACY_SERVICE environment variable.
wandb: Currently logged in as: karl-find-hansen (karl-find-hansen-technical-university-of-denmark) to https://api.wandb.ai. Use `wandb login --relogin` to force relogin
wandb: WARNING Ignoring project 'greenland-bedmap-generation' when running a sweep.
wandb: Tracking run with wandb version 0.19.10
wandb: Run data is saved locally in /zhome/a1/4/145993/courses/special course/Special-Course/wandb/run-20250423_162752-a79wjaye
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run divine-sweep-2
wandb: ‚≠êÔ∏è View project at https://wandb.ai/karl-find-hansen-technical-university-of-denmark/greenland-bedmap-generation
wandb: üßπ View sweep at https://wandb.ai/karl-find-hansen-technical-university-of-denmark/greenland-bedmap-generation/sweeps/43odwsqx
wandb: üöÄ View run at https://wandb.ai/karl-find-hansen-technical-university-of-denmark/greenland-bedmap-generation/runs/a79wjaye

Aborted!
wandb: Ctrl-c pressed. Waiting for runs to end. Press ctrl-c again to terminate them.
Traceback (most recent call last):
  File "/appl9/python/3.13.0/lib/python3.13/threading.py", line 363, in wait
    gotit = waiter.acquire(True, timeout)
KeyboardInterrupt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/appl9/python/3.13.0/lib/python3.13/queue.py", line 213, in get
    self.not_empty.wait(remaining)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
  File "/appl9/python/3.13.0/lib/python3.13/threading.py", line 368, in wait
    self._acquire_restore(saved_state)
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/appl9/python/3.13.0/lib/python3.13/threading.py", line 315, in _acquire_restore
    def _acquire_restore(self, x):
    
KeyboardInterrupt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/wandb/__main__.py", line 3, in <module>
    cli.cli(prog_name="python -m wandb")
    ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/wandb/cli/cli.py", line 103, in wrapper
    return func(*args, **kwargs)
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/wandb/cli/cli.py", line 290, in service
    server.serve()
    ~~~~~~~~~~~~^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/wandb/sdk/service/server.py", line 106, in serve
    mux.loop()
    ~~~~~~~~^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/wandb/sdk/service/streams.py", line 426, in loop
    raise e
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/wandb/sdk/service/streams.py", line 424, in loop
    self._loop()
    ~~~~~~~~~~^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/wandb/sdk/service/streams.py", line 414, in _loop
    action = self._action_q.get(timeout=1)
  File "/appl9/python/3.13.0/lib/python3.13/queue.py", line 194, in get
    with self.not_empty:
         ^^^^^^^^^^^^^^
  File "/appl9/python/3.13.0/lib/python3.13/threading.py", line 307, in __exit__
    return self._lock.__exit__(*args)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^
RuntimeError: release unlocked lock
Traceback (most recent call last):
  File "/zhome/a1/4/145993/courses/special course/Special-Course/src/optimize_wandb.py", line 322, in <module>
    train()
    ~~~~~^^
  File "/zhome/a1/4/145993/courses/special course/Special-Course/src/optimize_wandb.py", line 47, in train
    dataset = ArcticDataloader(
        bedmachine_path=os.path.join("data","inputs", "Bedmachine", "BedMachineGreenland-v5.nc"),
    ...<3 lines>...
        hillshade_path=os.path.join("data", "inputs", "hillshade", "macgregortest_flowalignedhillshade.tif"),
    )
  File "/zhome/a1/4/145993/courses/special course/Special-Course/data/data_preprocessing.py", line 43, in __init__
    self.height_map_icecap_tensor = self.align_to_velocity(self.height_map_icecap_data)
                                    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/zhome/a1/4/145993/courses/special course/Special-Course/data/data_preprocessing.py", line 104, in align_to_velocity
    aligned = data.rio.reproject_match(self.ice_velocity_data['land_ice_surface_easting_velocity'])
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/rioxarray/raster_array.py", line 585, in reproject_match
    reprojected_data_array = self.reproject(
        match_data_array.rio.crs,
    ...<3 lines>...
        **reproject_kwargs,
    )
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/rioxarray/raster_array.py", line 489, in reproject
    source=self._obj.values,
           ^^^^^^^^^^^^^^^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/xarray/core/dataarray.py", line 814, in values
    return self.variable.values
           ^^^^^^^^^^^^^^^^^^^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/xarray/core/variable.py", line 507, in values
    return _as_array_or_item(self._data)
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/xarray/core/variable.py", line 301, in _as_array_or_item
    data = np.asarray(data)
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/xarray/core/indexing.py", line 509, in __array__
    return np.asarray(self.get_duck_array(), dtype=dtype, copy=copy)
                      ~~~~~~~~~~~~~~~~~~~^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/xarray/core/indexing.py", line 835, in get_duck_array
    self._ensure_cached()
    ~~~~~~~~~~~~~~~~~~~^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/xarray/core/indexing.py", line 832, in _ensure_cached
    self.array = as_indexable(self.array.get_duck_array())
                              ~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/xarray/core/indexing.py", line 789, in get_duck_array
    return self.array.get_duck_array()
           ~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/xarray/core/indexing.py", line 652, in get_duck_array
    array = self.array[self.key]
            ~~~~~~~~~~^^^^^^^^^^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/rioxarray/_io.py", line 456, in __getitem__
    return indexing.explicit_indexing_adapter(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        key, self.shape, indexing.IndexingSupport.OUTER, self._getitem
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/xarray/core/indexing.py", line 1013, in explicit_indexing_adapter
    result = raw_indexing_method(raw_key.tuple)
  File "/zhome/a1/4/145993/courses/special course/special_course/lib/python3.13/site-packages/rioxarray/_io.py", line 433, in _getitem
    out = riods.read(band_key, window=window, masked=self.masked)
  File "rasterio/_io.pyx", line 644, in rasterio._io.DatasetReaderBase.read
  File "rasterio/_io.pyx", line 969, in rasterio._io.DatasetReaderBase._read
  File "rasterio/_io.pyx", line 199, in rasterio._io.io_multi_band
  File "/appl9/python/3.13.0/lib/python3.13/contextlib.py", line 145, in __exit__
    def __exit__(self, typ, value, traceback):
    
KeyboardInterrupt
Terminated
